src/merge.ts

```diff
@@ -188,21 +189,25 @@ function getMinScoreId(bestMatchesIdToUnits: Map<string, { elem: XmlElement; sco
     return minScoreId;
 }
 
-export function mergeWithMapping(inFileContent: string, destFileContent: string, options?: MergeOptions, destFilePath?: string): [mergedDestFileContent: string, idMappging: { [oldId: string]: string }] {
-    const inDoc = new XmlDocument(inFileContent);
-    const xliffVersion = inDoc.attr.version as '1.2' | '2.0' ?? '1.2';
+export function mergeWithMapping(inFilesContent: string | string[], destFileContent: string, options?: MergeOptions, destFilePath?: string): [mergedDestFileContent: string, idMappging: { [oldId: string]: string }] {
+    inFilesContent = Array.isArray(inFilesContent) ? inFilesContent : [inFilesContent];
+    const inDocs = inFilesContent.map(inFileContent => new XmlDocument(inFileContent));
+    const xliffVersion = inDocs[0].attr.version as '1.2' | '2.0' ?? '1.2';
 
-    const destDoc = new XmlDocument(destFileContent.match(/^[\n\r\s]*$/) ? createEmptyTarget(xliffVersion === '2.0', extractSourceLocale(inDoc, xliffVersion === '2.0'), extractTargetLocale(destFilePath)) : destFileContent);
+    const destDoc = new XmlDocument(destFileContent.match(/^[\n\r\s]*$/) ? createEmptyTarget(xliffVersion === '2.0', extractSourceLocale(inDocs[0], xliffVersion === '2.0'), extractTargetLocale(destFilePath)) : destFileContent);
+    const excludeDocs = (options?.excludeFiles ?? []).map(excludeFile => new XmlDocument(excludeFile));
 
     const destUnitsParent = xliffVersion === '2.0' ? destDoc.childNamed('file')! : destDoc.childNamed('file')?.childNamed('body')!;
-    const inUnits = getUnits(inDoc, xliffVersion) ?? [];
+    const inUnits = inDocs.map(inDoc => getUnits(inDoc, xliffVersion) ?? []).flat(1);
     const inUnitsById = new Map<string, XmlElement>(inUnits.map(unit => [unit.attr.id!, unit]));
     const destUnitsById = new Map<string, XmlElement>((getUnits(destDoc, xliffVersion) ?? []).map(unit => [unit.attr.id!, unit]));
+    const excludeUnits = excludeDocs.map(excludeDoc => getUnits(excludeDoc, xliffVersion) ?? []).flat(1);
+    const excludeUnitsById = new Map<string, XmlElement>(excludeUnits.map(unit => [unit.attr.id!, unit]));
     const allInUnitsWithoutDestinationUnit = inUnits.filter(u => !destUnitsById.has(u.attr.id));
     const allInUnitsWithDestinationUnit = inUnits.filter(u => destUnitsById.has(u.attr.id));
 
     // collect (potentially) obsolete units (defer actual removal to allow for fuzzy matching..):
-    const removeNodes = getUnits(destDoc, xliffVersion)!.filter(destUnit => !inUnitsById.has(destUnit.attr.id));
+    const removeNodes = getUnits(destDoc, xliffVersion)!.filter(destUnit => !inUnitsById.has(destUnit.attr.id) || excludeUnitsById.has(destUnit.attr.id));
```

Comment:
is it possible that this leads to situations where excludeUnits could be fuzzy matched (see above comment) in than be included again?

Reply:
Indeed I change it. The behaviour should be the same as in the removeIdsWithPrefix solution, i.e. as if the input didn't even have the excluded nodes.